block-producer/src/batch_builder/batch.rs

```diff
@@ -27,9 +28,10 @@ pub struct TransactionBatch {
     id: BatchId,
     updated_accounts: BTreeMap<AccountId, AccountStates>,
     produced_nullifiers: Vec<Nullifier>,
-    created_notes_smt: BatchNoteTree,
     /// The notes stored `created_notes_smt`
-    created_notes: Vec<NoteEnvelope>,
+    created_notes_smt: BatchNoteTree,
+    created_notes: Vec<NoteCreated>,
+    created_note_envelopes: Vec<NoteEnvelope>,
```

Comment:
Do we use these envelopes anywhere else?

Reply:
With the new refactor we can directly use the `created_note_envelopes_with_details` everywhere.